From 131b98eeed766a6f3d57cdf5b2ed7d4d3f650184 Mon Sep 17 00:00:00 2001
From: Sultanxda <sultanxda@gmail.com>
Date: Sun, 4 Oct 2015 00:20:01 -0700
Subject: [PATCH 081/173] SnapdragonCamera: Always lock AE and AWB when
 auto-focus is used

Lock the auto exposure (AE) and auto white balance (AWB) when tap-to-focus
is used.

Change-Id: I9d8e9b4928516e8b2bad7be958e5c4849db19ab6
Signed-off-by: Sultanxda <sultanxda@gmail.com>
---
 src/com/android/camera/FocusOverlayManager.java | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/com/android/camera/FocusOverlayManager.java b/src/com/android/camera/FocusOverlayManager.java
index 5313e0ed4..00e7a535a 100644
--- a/src/com/android/camera/FocusOverlayManager.java
+++ b/src/com/android/camera/FocusOverlayManager.java
@@ -285,6 +285,9 @@ public class FocusOverlayManager {
             // sound.
             if (focused) {
                 mState = STATE_SUCCESS;
+                // Lock exposure and white balance
+                setAeAwbLock(true);
+                mListener.setFocusParameters();
             } else {
                 mState = STATE_FAIL;
             }
@@ -296,6 +299,9 @@ public class FocusOverlayManager {
             // take the picture now.
             if (focused) {
                 mState = STATE_SUCCESS;
+                // Lock exposure and white balance
+                setAeAwbLock(true);
+                mListener.setFocusParameters();
             } else {
                 mState = STATE_FAIL;
             }
@@ -448,6 +454,7 @@ public class FocusOverlayManager {
         // Otherwise, focus mode stays at auto and the tap area passed to the
         // driver is not reset.
         resetTouchFocus();
+        setAeAwbLock(false);
         mListener.cancelAutoFocus();
         mUI.resumeFaceDetection();
         mState = STATE_IDLE;
-- 
2.17.1

